docs/rules/prefer-destructuring.md

```diff
@@ -29,11 +29,15 @@ var foo = object.bar;
 
 ### Options
 
-This rule takes two sets of configuration objects; the first controls the types that the rule is applied to, and the second controls the way those objects are evaluated.
+This rule takes two sets of configuration objects. The first object parameter determines what types of destructuring the rule applies to.
 
-The first has two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently.  By default, both are `true`.
+The two properties, `array` and `object`, can be used to turn on or off the destructuring requirement for each of those types independently. By default, both are true.
```

Comment:
I think it would be useful to give more examples for each option format after these descriptions, e.g.

````markdown
The two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently. By default, both are true.

For example, the following configuration enforces only object destructuring, but not array destructuring:

```json
{
  "rules": {
    "prefer-destructuring": ["error", {"object": true, "array": false}]
  }
}
```
````